src/Components/RecommendationsPage/RecommendationsTable.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { useContext, useEffect, useState } from 'react';
+import { Bullseye, Button } from '@patternfly/react-core';
+import { EmptyState, EmptyStateHeader, EmptyStateIcon, EmptyStateVariant } from '@patternfly/react-core';
+import { SearchIcon } from '@patternfly/react-icons';
+import { Table, Tbody, Td, Th, Thead, ThProps, Tr } from '@patternfly/react-table';
+import { DateFormat } from '@redhat-cloud-services/frontend-components/DateFormat';
+
+import { getRecommendations, isError } from '../../api';
+import { RECOMMENDATIONS_DETAIL_ROOT } from '../../Helpers/constants';
+import { loadingSkeletons } from '../../Helpers/Helpers';
+import { displayErrorNotification, sortTable } from '../../Helpers/Helpers';
+import { RegistryContext } from '../../store';
+import { Recommendation } from './types';
+
+const RecommendationsTable = ({ page, perPage, setTotal }) => {
+  const columnNames = {
+    description: 'Name',
+    publish_date: 'Modified',
+    systems: 'Systems',
+    remediation: 'Remediation',
+    run: 'Run',
+  };
+
+  const [showSkeleton, setShowSkeleton] = useState(false);
+  const [activeSortIndex, setActiveSortIndex] = React.useState<number | null>(null);
+  const [activeSortDirection, setActiveSortDirection] = React.useState<'asc' | 'desc' | null>(null);
+  const [recommendations, setRecommendations] = React.useState<Recommendation[]>([]);
+
+  const { getRegistry } = useContext(RegistryContext);
+
+  const buildSort = (direction, index): string => {
+    if (direction === null || index === null) {
+      return 'description';
+    }
+    const sortMap = {
+      0: 'description',
+      1: 'publish_date',
+      2: 'impacted_count',
+      3: 'playbook_count',
+    };
+    const sign = direction === 'asc' ? '' : '-';
+    const sort = sign + sortMap[index];
+    return sort;
+  };
+
+  const fetchData = async (page: number, perPage: number, sort) => {
+    setShowSkeleton(true);
+
+    const response = await getRecommendations(page, perPage, sort);
+    if (isError(response)) {
+      const store = getRegistry().getStore();
+      displayErrorNotification(store, response.message);
+      return;
+    }
+
+    let recommendations = response.data?.data;
+    recommendations = recommendations?.map((rec) => ({
+      id: rec.rule_id,
+      description: rec.description,
+      publish_date: rec.publish_date,
+      systems: rec.impacted_systems_count,
+      remediation: rec.playbook_count > 0 ? 'Playbook' : 'Manual',
+    }));
+    setRecommendations(recommendations);
+
+    const count = response.data?.meta?.count;
+    setTotal(count);
+
+    setShowSkeleton(false);
+    return count;
+  };
+
+  const getSortableRowValues = (rec: Recommendation): (string | number | Date)[] => {
+    const { description, publish_date, systems, remediation } = rec;
+    return [description, publish_date, systems, remediation];
+  };
+
+  sortTable(recommendations, activeSortIndex, activeSortDirection, getSortableRowValues);
```

Comment:
I would expect to call `sortTable` on some sort button click in table, and also why there is data refetch in `useEffect` based on `activeSortDirection, activeSortIndex` when you sort only locally and don't get sorted data from BE.

Reply:
rn the change in `activeSortDirection, activeSortIndex` makes an api call, therefore they are in the useEffect dependencies
the `sortTable`  in the component body is taken from the patternfly examples, because sorting is done by the BE, im removing it